### ISSUE TITLE
[Bug]: Import CSV and "SAVE AS" problem in  "zh_Hant" interface. JASP version 0.18.1

### ISSUE THREAD

**Issue by akwan168** [jasp-stats/jasp-issues#2311]
### JASP Version

0.18.0

### JASP Module

Unrelated

### What analysis are you seeing the problem on?

import & save file

### What OS are you seeing the problem on?

Windows 10

### Bug Description

1. When the data file in csv format contains Chinese variables, garbled characters will appear when imported into JASP, regardless of the English interface or the "zh_Hant" interface.

2. In the English interface, there is no problem using "SAVE" or "SAVE AS". However, in the "zh_Hant" interface, although there is no problem using "SAVE", using "SAVE AS" seems to cause some bugs.

3. I have videotaped the operation process, please see the video below:
    https://youtu.be/1-keTSfbKDU

### Expected Behaviour

please see the video below: https://youtu.be/1-keTSfbKDU

### Steps to Reproduce

1.the bug is reported in following video：https://youtu.be/1-keTSfbKDU
2.
3.
...
[test.zip](https://github.com/jasp-stats/jasp-issues/files/12769782/test.zip)

### Final Checklist

- [X] I have included a screenshot showcasing the issue, if possible.
- [X] I have included a JASP file (zipped) or data file that causes the crash/bug, if applicable.
- [X] I have accurately described the bug, and steps to reproduce it.

**Comment by shun2wang** [comment]
this was duplicated with https://github.com/jasp-stats/jasp-issues/issues/2289 and fixed in 0.18.1, can you try the new release? thanks.

im closing it for now if you still have problem please feel free to reopen it.

**Comment by akwan168** [comment]
Hello!
I did test it in jasp 0.18.1. The bug I responded to is obviously different
from 2289.
How can I reopen it

Thanks

=======================================
陳寬裕
Chen, Kuan-Yu
屏東科技大學休閒運動健康系(行銷、觀光休閒領域)  教授

0933688420
========================================


Shun Wang ***@***.***> 於 2023年9月30日 週六 上午6:31寫道：

**Comment by shun2wang** [comment]
okay, seems really different. thanks for your report.

**Comment by shun2wang** [comment]
Hi @akwan168 For the **bug1**: Can you try set "Preference" -> "Data" -> "Windows workaround" and then check the "Assum CSV..." -> choose "950 (ANSI/OEM -繁体中文 Big5)" encoding? or something others has `Big5`  encoding available on your PC.I think this not a bug because csv exported from Excel will not be encoded as UTF-8 but a locale encode like `big5` for traditional Chinese in your file.

For the **bug2**: @boutinb it may because here (`JASP 檔 （*.jasp）` at https://hosted.weblate.org/translate/jasp/jaspdesktop/zh_Hant/?checksum=f836cb3ee97de466) using a full-width bracketsa  character in the file suffix in the explorer ( I cannot change it due to locked repo ) so will auto added a `）` in the filename. I think we can change the code not to extract the suffix from the translated characters to avoid such errors?

**Comment by akwan168** [comment]
Hi

Thank you for your reply.
Regarding bug1, the problem can indeed be fixed according to the method you
provided.
As for bug2, I look forward to the arrival of the next version.

thank you for your enthusiasm

=======================================
Chen, Kuan-Yu
Professor, Department of Leisure Sports and Health,
Pingtung University of Science and Technology
========================================


Shun Wang ***@***.***> 於 2023年10月7日 週六 上午10:10寫道：

**Comment by boutinb** [comment]
> ( I will change it after weblate unlocked )

Weblate is finally unlocked...

**Comment by shun2wang** [comment]
> Weblate is finally unlocked...

I saw and changed.

>I think we can change the code not to extract the suffix from the translated characters to avoid such errors?

@boutinb What do you think?

**Comment by boutinb** [comment]
> > I think we can change the code not to extract the suffix from the translated characters to avoid such errors?
> 
> @boutinb What do you think?
Yes the translated string should be only "JASP Files" and not "JASP Files (*.jasp)". The " (*.jasp)" should be added afterwards to the filter string.

**Comment by akwan168** [comment]
I think we can change the code not to extract the suffix from the
translated characters to avoid such errors?

@boutinb <https://github.com/boutinb> What do you think?

ok ！ Thanks!

=======================================
陳寬裕
Chen, Kuan-Yu
屏東科技大學休閒運動健康系(行銷、觀光休閒領域)  教授

0933688420
========================================


Shun Wang ***@***.***> 於 2023年10月9日 週一 下午8:41寫道：
